src/components/ItemsList.js

```diff
@@ -9,25 +9,34 @@ const ItemsList = () => {
   );
   const history = useHistory();
 
-  // made this function just for testing purposes so it's easier to clear
-  // the token and go back to the home page
   const removeToken = () => {
     localStorage.removeItem('token');
     history.push('/');
   };
 
+  const addItem = () => {
+    history.push('/add');
+  };
+
   return (
     <div>
       {loading && <>Loading</>}
       {error && <>Error</>}
+      <h1>Collection:</h1>
       {snapshot && (
         <>
-          <h1>Collection:</h1>
-          <ul>
-            {snapshot.docs.map((doc, index) => (
-              <li key={index}>{doc.data().name}</li>
-            ))}
-          </ul>
+          {!snapshot.docs.length ? (
```

Comment:
Sometime, it can get confusing to have nested conditionals like this. It's actually possible to split up the checks here so that they are not nested, using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining):

```
{snapshot?.docs.length > 0 && (
 // render the list of items
)}

{snapshot?.docs.length === 0 && (
 // shopping list empty
)}
```

Reply:
Thanks for the suggestion @luisaugusto, yeah we also thought about nested conditionals and first created the condition outside but we had a problem, it was showing the welcome prompt while loading the page and disappearing immediately, and can you tell me about `snapshot?`. Why do we need `?`?

Reply:
@hulyak the `?.` is used for Optional Chaining, which allows you to check for object values before traversing deeper down in the object. [This article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) has some good information, but it allows you to turn `snapshot && snapshot.docs.length` into `snapshot?.docs.length`

Reply:
Great, thank you @luisaugusto